### ISSUE TITLE
Readability, Class Pertinence, and Documentation

### ISSUE THREAD

**Issue by elawrie** [kilometersvi/EnergySymbolicRegression#2]
In reviewing your project, I found it interesting but there is a lack of use case and pertinence to our class's concepts. I don't see how this represents a DSL and the documentation provided is confusing, especially if the reader has no background in regression, neural networks, or theoretical machine learning as a whole. I understand how you are editing the loss function from the Hopfield neural network formula and using energy instead of data to perform and evaluate regression, but it seems like your code contains a lot of different components with little information about how they work together to perform your desired tasks. Your code lacks comments and readability that would allow us to review your project. I would second Alex's issue raised on documentation, as your peer review folders contain no information on how to run or possibly add features to your project (I only see source code). It would be great to see a very basic outline of what the goal of your project is and how to interact with it.

**Comment by kilometersvi** [comment]
Thanks for the criticism Evelyn. 

The goal of this project is to create a DSL for explainable machine learning. Machine learning in its current implementation is a "black box," taking advantage of random connections, etc in order to output apparent results. Although functional, it is difficult to design models for use cases more specific than knowing data type and data structure, and trusting the model to determine the distribution (which is the design of these models). The core model implemented here, a Hopfield model, is explainable in that it is an RNN who's weights are intentionally set by the user based on some mathematical understanding of the desired behavior. This is done through adjustments to the connectivity matrix, or Q matrix. The end goal of this project is to create a DSL for adjusting the Q matrix; but in order to do that, the base model that the DSL will operate on must be created first.

(Symbolic regression is one such implementation for this model, which I am focusing on for simplicity and maintaining some sort of focus during development.)

In regards to readability and documentation: Documentation will be added with papers referenced for choices regarding different computations. Project state is still fairly exploratory and using unconventional methods; after completion of loss calculation, repository will be reorganized to better illustrate functionality and increase readability. Right now, it is for sure a mess.

Apologies for the lack of documentation by the time of a peer review. Issue will be closed after additional documentation + peer review material pushed.
